### PR TITLE
Allow lime dependencies between targets and document these

### PR DESCRIPTION
**Pull request by Dschoordsch**
Allow adding sources only from specific source set, main or common. Also pass interface sources as auxiliary lime files to allow lime dependencies between targets. Add minimal documentation for these.

### CODE REVIEW

File: examples/cmake/README.md [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
 This project provides CMake functions to simplify using Gluecodium in CMake-based
 projects.
+
+## Generate code
+
+The modules consist of a platform independent part in `modules/gluecodium/gluecodium`
+and platform specific helpers. To generate code for a target the functions
+`apigen_generate`, `apigen_target_sources` and `apigen_target_include_directories`
+should be called for it.
+Depending on the target platform, there are helpers for `.aar` archives for Android  and
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: "there are helpers for **creating**" would read better. Also there's an extraneous whitespace between "Android" and "and".

File: examples/cmake/README.md [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
-
 This project provides CMake functions to simplify using Gluecodium in CMake-based
 projects.
+
+## Generate code
+
+The modules consist of a platform independent part in `modules/gluecodium/gluecodium`
+and platform specific helpers. To generate code for a target the functions
+`apigen_generate`, `apigen_target_sources` and `apigen_target_include_directories`
+should be called for it.
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: "should be called for that target". The sentence is rather long and it's easy to forget what 'it' refers to by the end of it.

File: examples/cmake/README.md [hunk 3]
```diff
@@ -1,6 +1,44 @@
...
+Depending on the target platform, there are helpers for `.aar` archives for Android  and
+Apples `.framework` for Swift.
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: "Apple" or even "iOS"

File: examples/cmake/README.md [hunk 4]
```diff
@@ -1,6 +1,44 @@
...
+For more details check out the [Examples](../examples/README.md).
+
+## Source sets
+
+Gluecodium distinguishes between two different source sets, "main" and "common". "Main"
+code depends on the input Lime files while "common" code only depends on the Gluecodium
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: "Lime IDL" instead of "Lime". Or even just "IDL".

File: examples/cmake/README.md [hunk 5]
```diff
@@ -1,6 +1,44 @@
...
+be combined with the main code and can be used like this.
+
+If code is generated for more than one target and should be used together, then the
+common code needs to be separated as it would otherwise be duplicated and cause
+compilation issues. For this a separate common output directory must be set for
+`apigen_generate` and the generated common source only added to one target which all
+others link against. `add_target_sources` and `add_target_link_directories` support
+"MAIN" and "COMMON" options for this use case.
+
+## Lime file dependencies between targets
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: Same as above

File: examples/cmake/README.md [hunk 6]
```diff
@@ -1,6 +1,44 @@
...
+
+If a target A's lime files depend on the lime files of target B, then these need to
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: Same as above

File: examples/cmake/README.md [hunk 7]
```diff
@@ -1,6 +1,44 @@
...
+be passed to Gluecodium during generation of A. To do this without generating B's
+code twice and having duplicate symbols, Gluecodium supports auxiliary lime input
```
Comment (DanielKamkha) [inline examples/cmake/README.md]: Same as above

File: examples/cmake/modules/gluecodium/gluecodium/Generate.cmake [hunk 8]
```diff
@@ -192,6 +192,8 @@ function(apigen_generate)
         -DAPIGEN_COMMON_OUTPUT_DIR=${APIGEN_COMMON_OUTPUT_DIR}
         -DAPIGEN_BUILD_OUTPUT_DIR=${APIGEN_BUILD_OUTPUT_DIR}
         -DAPIGEN_VERBOSE=${APIGEN_VERBOSE}
+	# Pass on the interface lime files from dependencies, only INTERFACE_SOURCES is propagated trinsitively as of CMake 3.16
```
Comment (DanielKamkha) [inline examples/cmake/modules/gluecodium/gluecodium/Generate.cmake]: "trinsitively" should be "transitively"

File: examples/cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake [hunk 9]
```diff
@@ -35,50 +35,60 @@ cmake_minimum_required(VERSION 3.5)
...
 # The general form of the command is::
 #
-#     apigen_target_include_directories(target [ALWAYS_EXPOSE_CPP])
-#
-# The ALWAYS_EXPOSE_CPP argument may be used to expose the C++ headers in all situations. This can
-# be useful when multiple Gluecodium-generated libraries depend on each-other.
+#     apigen_target_include_directories(target [MAIN] [COMMON])
+#        <target>    Target for which source was generated via `apigen_generate`
+#        MAIN        Add the include path to the MAIN generated source set, i.e.
+#                    code generated for the input Lime files.
```
Comment (DanielKamkha) [inline examples/cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake]: "Lime" -> "Lime IDL"

File: examples/cmake/modules/gluecodium/gluecodium/TargetSources.cmake [hunk 10]
```diff
@@ -35,18 +35,39 @@ cmake_minimum_required(VERSION 3.5)
 #
 # The general form of the command is::
 #
-#     apigen_target_sources(target)
+#     apigen_target_sources(<target> (MAIN) (COMMON))
+#        <target>    Target for which source was generated via `apigen_generate`
+#        MAIN        Add the MAIN generated source set, i.e. code generated for
+#                    the input Lime files.
```
Comment (DanielKamkha) [inline examples/cmake/modules/gluecodium/gluecodium/TargetSources.cmake]: Same as above

File: examples/cmake/modules/gluecodium/gluecodium/TargetSources.cmake [hunk 11]
```diff
@@ -35,18 +35,39 @@ cmake_minimum_required(VERSION 3.5)
...
+#        COMMON      Add the common generated source set which is independent of
+#                    input lime files and can be shared between different targets
```
Comment (DanielKamkha) [inline examples/cmake/modules/gluecodium/gluecodium/TargetSources.cmake]: Same as above

File: cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake [hunk 12]
```diff
@@ -35,50 +35,60 @@ cmake_minimum_required(VERSION 3.5)
...
-
+    if(${GENERATOR} MATCHES android)
+      target_include_directories(${target}
+        PUBLIC $<BUILD_INTERFACE:${COMMON_OUTPUT_DIR}/android/jni>)
+    endif()
   endif()
 
+  if(${GENERATOR} MATCHES android)
+    # Check if we are doing a host build (no cross compilation)
+    if(NOT(${CMAKE_SYSTEM_NAME} STREQUAL "Android"))
```
Comment (Hsilgos) [inline cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake]: There is variable [CMAKE_CROSSCOMPILING](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html) which you can check instead.

Comment (Hsilgos) [reply cmake/modules/gluecodium/gluecodium/TargetIncludeDirectories.cmake]: I mean instead of ${CMAKE_SYSTEM_NAME} STREQUAL "Android"

**Pull request merged**